hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/RexToExpression.java

```diff
@@ -17,39 +17,272 @@
+            case LESS_THAN:
+                return ComparisonPredicate.create(operands[0], operands[1], ComparisonMode.LESS_THAN);
+
+            case LESS_THAN_OR_EQUAL:
+                return ComparisonPredicate.create(operands[0], operands[1], ComparisonMode.LESS_THAN_OR_EQUAL);
+
+            case IS_TRUE:
+                return IsTruePredicate.create(operands[0]);
+
+            case IS_NOT_TRUE:
+                return IsNotTruePredicate.create(operands[0]);
+
+            case IS_FALSE:
+                return IsFalsePredicate.create(operands[0]);
+
+            case IS_NOT_FALSE:
+                return IsNotFalsePredicate.create(operands[0]);
+
             case IS_NULL:
                 return IsNullPredicate.create(operands[0]);
 
+            case IS_NOT_NULL:
+                return IsNotNullPredicate.create(operands[0]);
+
             default:
                 break;
         }
 
         throw QueryException.error("Unsupported operator: " + operator);
     }
+
+    private static Expression<?> convertBooleanLiteral(RexLiteral literal, SqlTypeName type) {
+        assert type == SqlTypeName.BOOLEAN;
+        Boolean value = literal.getValueAs(Boolean.class);
+        return ConstantExpression.create(value, SqlToQueryType.map(type));
+    }
+
+    private static Expression<?> convertNumericLiteral(RexLiteral literal, SqlTypeName type) {
+        Object value;
+        switch (type) {
+            case TINYINT:
+                value = literal.getValueAs(Byte.class);
+                break;
+
+            case SMALLINT:
+                value = literal.getValueAs(Short.class);
+                break;
+
+            case INTEGER:
+                value = literal.getValueAs(Integer.class);
+                break;
+
+            case BIGINT:
+                value = literal.getValueAs(Long.class);
+                break;
+
+            case DECIMAL:
+                value = literal.getValueAs(BigDecimal.class);
+                break;
+
+            case REAL:
+                value = literal.getValueAs(Float.class);
+                break;
+
+            case DOUBLE:
+                value = literal.getValueAs(Double.class);
+                break;
+
+            default:
+                throw new IllegalArgumentException("Unsupported literal type: " + type);
+        }
+
+        return ConstantExpression.create(value, SqlToQueryType.map(type));
+    }
+
+    private static Expression<?> convertStringLiteral(RexLiteral literal, SqlTypeName type) {
+        Object value;
+        switch (type) {
+            case CHAR:
+            case VARCHAR:
+                value = literal.getValueAs(String.class);
+                break;
+
+            default:
+                throw new IllegalArgumentException("Unsupported literal type: " + type);
+        }
+
+        return ConstantExpression.create(value, SqlToQueryType.map(type));
+    }
+
+    private static Expression<?> convertTemporalLiteral(RexLiteral literal, SqlTypeName type) {
```

Comment:
This part of the class is not tested at all, therefore we do not know whether it works or not. Given that this PR is not concerned with temporal types, we'd better not to merge this code at all.

Reply:
fixed in 1a8e2fa6192f905d7c9293f7081796bf21433c06